File: minesweeper/java-minesweeper/src/test/java/it/chicio/minesweeper/field/parser/FieldRowContentTerminationChainParserTest.java
package it.chicio.minesweeper.field.parser;

import it.chicio.minesweeper.field.Field;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class FieldRowContentTerminationChainParserTest {
    private FieldRowContentTerminationChainParser fieldRowContentTerminationChainParser;

    @Before
    public void setUp() {
        fieldRowContentTerminationChainParser = new FieldRowContentTerminationChainParser();
    }

    @Test
    public void canParseTerminationRow() {
        assertThat(fieldRowContentTerminationChainParser.canParse("0 0"), is(true));
    }

    @Test
    public void canNotParseTerminationRow() {
        assertThat(fieldRowContentTerminationChainParser.canParse("0"), is(false));
        assertThat(fieldRowContentTerminationChainParser.canParse("* *"), is(false));
    }

    @Test
    public void parseValidTerminationRow() {
        Field currentField = new Field(new String[][]{{"*"}});
        FieldsParsingStatus fieldsParsingStatus = fieldRowContentTerminationChainParser.parse(new FieldsParsingStatusBuilder()
                .withCurrentField(currentField)
                .withFieldsParsed(new ArrayList<Field>())
                .withCurrentRow(1)
                .build()
        );

        assertThat(fieldsParsingStatus.fieldsParsed.size(), is(1));
        assertThat(fieldsParsingStatus.fieldsParsed.get(0), is(equalTo(currentField)));
    }
}

File: minesweeper/kotlin-mockk-minesweeper/src/test/kotlin/it/chicio/minesweeper/field/parser/FieldRowContentTerminationChainParserTest.kt
package it.chicio.minesweeper.field.parser

import it.chicio.minesweeper.FieldFactory
import it.chicio.minesweeper.FieldsParsingStatusBuilder
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("FieldRowContentTerminationChainParser")
class FieldRowContentTerminationChainParserTest {
    private lateinit var fieldRowContentTerminationChainParser: FieldRowContentTerminationChainParser
    @BeforeEach
    fun setUp() {
        fieldRowContentTerminationChainParser = FieldRowContentTerminationChainParser()
    }

    @Test
    fun `can parse termination row`() {
        assertTrue(fieldRowContentTerminationChainParser.canParse("0 0"))
    }

    @Test
    fun `can not parse termination row`() {
        assertFalse(fieldRowContentTerminationChainParser.canParse("0"))
        assertFalse(fieldRowContentTerminationChainParser.canParse("* *"))
    }

    @Test
    fun `parse valid termination row`() {
        val fieldsParsingStatus = fieldRowContentTerminationChainParser.parse(FieldsParsingStatusBuilder(
                currentField = FieldFactory().make(arrayOf(arrayOf("*"))),
                currentRow = 1
        ).build())

        assertEquals(fieldsParsingStatus.fieldsParsed.size, 1)
        assertEquals(fieldsParsingStatus.fieldsParsed[0], FieldFactory().make(arrayOf(arrayOf("*"))))
    }
}